### Create either a python script or a jupyter notebook named explore_tips that explores the tips data set that is built in to seaborn. Perform at least 1 t-test and 1 chi square test.

In [10]:
import pandas as pd
import seaborn as sns
from scipy import stats

In [8]:
df = sns.load_dataset('tips')
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [9]:
tips_dummies = pd.get_dummies(df[['sex', 'smoker', 'day', 'time']], drop_first=True)
df = pd.concat([df, tips_dummies], axis=1)
df.head()

total_bill   tip     sex smoker  day    time  size  sex_Female  smoker_No  \
0       16.99  1.01  Female     No  Sun  Dinner     2           1          1   
1       10.34  1.66    Male     No  Sun  Dinner     3           0          1   
2       21.01  3.50    Male     No  Sun  Dinner     3           0          1   
3       23.68  3.31    Male     No  Sun  Dinner     2           0          1   
4       24.59  3.61  Female     No  Sun  Dinner     4           1          1   

   day_Fri  day_Sat  day_Sun  time_Dinner  
0        0        0        1            1  
1        0        0        1            1  
2        0        0        1            1  
3        0        0        1            1  
4        0        0        1            1

### T-Test

$H_0$: Tip ammount is unaffected by whether or not it is dinner or lunch.  
$H_a$: Tip ammount is higher during dinner rather than lunch.  
alpha = .05

In [11]:
x1 = df[df.time_Dinner == 1].tip
x2 = df[df.time_Dinner == 0].tip

In [12]:
stats.ttest_ind(x1, x2)

Ttest_indResult(statistic=1.9062569301202392, pvalue=0.05780153475171558)

Our P-value comes out to .0578 which is very close but doesn't quite go lower than our set alpha. This mean that while they aren't neccesarily correlated it is likely that there is a weak connection there between the two.

### Chi Square

$H_0$: Female diners are not more likely to have either lunch or dinner.  
$H_0$: Female diners are more likely to have dinner than lunch.
alpha = .05

In [26]:
observed = pd.crosstab(df.sex_Female, df.time_Dinner)
observed

time_Dinner   0    1
sex_Female          
0            33  124
1            35   52

In [27]:
chi2, p, degf, expected = stats.chi2_contingency(observed)

In [28]:
print(chi2, p)

9.343808982970623 0.002237400118075248


Our P-value comes out to .002 which is smaller than our alpha of .05. This means that the female diners are more likely to eat dinner at this establishment than lunch.